### PR TITLE
Move URL escapify functions into tscore

### PR DESCRIPTION
**Pull request by maskit**
The code for escaping URL is currently in `proxy/logging` but it's not logging specific and it should be useful for other modules. This PR moves the logic into tscore, and add some unit tests.

I have WIP code for OCSP GET request, which is in `iocore/net`, and it relies on this change (URL escapify is needed to send OCSP request as a query string).

Co-Authored-By: Jeff Elsloo <elsloo@apache.org>

### CODE REVIEW

File: proxy/logging/unit-tests/test_LogUtils.cc [hunk 1]
```diff
@@ -150,52 +150,4 @@ TEST_CASE("get_unrolled_filename parses possible log files as expected", "[get_u
...
-  char output[128];
-  int output_len;
-
-  int n = sizeof(input) / sizeof(input[0]);
-  for (int i = 0; i < n; ++i) {
-    LogUtils::escapify_url(NULL, input[i], std::strlen(input[i]), &output_len, output, 128);
-    CHECK(std::string_view(output) == expected[i]);
-  }
-}
+}
```
Comment (bneradt) [inline proxy/logging/unit-tests/test_LogUtils.cc:153]: Github shows that this file doesn't have a trailing newline. Might want to update your editor to add those.

Comment (maskit) [reply proxy/logging/unit-tests/test_LogUtils.cc:153]: Do we require a trailing newline? I wonder if clang-format can correct it.

Comment (maskit) [reply proxy/logging/unit-tests/test_LogUtils.cc:153]: clang-format can do it, but we don't require it.
```
$ git diff .clang-format
diff --git a/.clang-format b/.clang-format
index 7931e13cf..d6431535b 100644
--- a/.clang-format
+++ b/.clang-format
@@ -127,7 +127,7 @@ IndentRequiresClause: true
 IndentWidth:     2
 IndentWrappedFunctionNames: false
 InsertBraces:    false
-InsertNewlineAtEOF: false
+InsertNewlineAtEOF: true
 InsertTrailingCommas: None
 IntegerLiteralSeparator:
   Binary:          0
```
then
```
$ git diff --stat
 .clang-format                                      | 2 +-
 include/shared/rpc/RPCClient.h                     | 2 +-
 include/tscore/Encoding.h                          | 2 +-
 mgmt/rpc/handlers/common/ErrorUtils.h              | 2 +-
 mgmt/rpc/handlers/common/Utils.h                   | 2 +-
 mgmt/rpc/handlers/common/convert.h                 | 2 +-
 mgmt/rpc/handlers/config/Configuration.h           | 2 +-
 mgmt/rpc/server/CommBase.cc                        | 2 +-
 mgmt/rpc/server/CommBase.h                         | 2 +-
 plugins/experimental/slice/prefetch.h              | 2 +-
 plugins/server_push_preload/server_push_preload.cc | 2 +-
 proxy/logging/unit-tests/test_LogUtils.cc          | 2 +-
 src/traffic_ctl/PrintUtils.h                       | 2 +-
 src/traffic_server/RpcAdminPubHandlers.h           | 2 +-
 src/tscore/Encoding.cc                             | 2 +-
 tests/tools/plugins/cert_update.cc                 | 2 +-
 16 files changed, 16 insertions(+), 16 deletions(-)
```

Comment (maskit) [reply proxy/logging/unit-tests/test_LogUtils.cc:153]: Added new line, and made a PR to require new line at eof.
https://github.com/apache/trafficserver/pull/9930

**Review by bneradt — APPROVED** [review]

**Pull request merged**